agentendpoint/config_task.go

```diff
@@ -234,16 +242,21 @@ func (c *configTask) postCheckState(ctx context.Context) {
 	// No prepopulate run for post check as we will always check every resource.
 	for i, osPolicy := range c.Task.GetOsPolicies() {
 		ctx = clog.WithLabels(ctx, map[string]string{"config_assignment": osPolicy.GetOsPolicyAssignment(), "policy_id": osPolicy.GetId()})
-		plcy := c.policies[osPolicy.GetId()]
-
+		plcy, ok := c.policies[osPolicy.GetId()]
+		if !ok {
+			continue
+		}
 		// Skip state check if this policy already has an error from a previous step.
 		if plcy.hasError {
 			clog.Debugf(ctx, "Policy has error, skipping post check.")
 			continue
 		}
-
 		pResult := c.results[i]
 		for i, configResource := range osPolicy.GetResources() {
+			res, ok := plcy.resources[configResource.GetId()]
+			if !ok || res == nil {
```

Comment:
Won't we return every resource in the response even if they are not executed? In that case, would we ever hit this condition?

Reply:
Same as above, no we should never hit this but a bug in the code that somehow gets past tests would cause a agent crash loop and we want to avoid that. I updated the code to log this error state

Reply:
/lgtm